pkg/madmin/config-commands.go

```diff
@@ -47,62 +43,45 @@ func (adm *AdminClient) GetConfig() ([]byte, error) {
 	return DecryptData(adm.secretAccessKey, resp.Body)
 }
 
-// GetConfigKeys - returns partial json or json value from config.json of a minio setup.
-func (adm *AdminClient) GetConfigKeys(keys []string) ([]byte, error) {
-	queryVals := make(url.Values)
-	for _, k := range keys {
-		queryVals.Add(k, "")
-	}
-
-	// Execute GET on /minio/admin/v1/config-keys to get config of a setup.
-	resp, err := adm.executeMethod("GET",
-		requestData{
-			relPath:     "/v1/config-keys",
-			queryValues: queryVals,
-		})
-	defer closeResponse(resp)
+// GetConfigKeyValue - get config entry, if no such
+// key exists, returns an error.
+func (adm *AdminClient) GetConfigKeyValue(input string) (KVS, error) {
+	configBytes, err := adm.GetConfig()
 	if err != nil {
 		return nil, err
 	}
-
-	if resp.StatusCode != http.StatusOK {
-		return nil, httpRespToErrorResponse(resp)
+	var m = make(ServerConfig)
+	if err = json.Unmarshal(configBytes, &m); err != nil {
+		return nil, err
 	}
-
-	return DecryptData(adm.secretAccessKey, resp.Body)
+	return getServerConfig(input, m)
```

Comment:
So we are dropping the encryption implementation that we had before? I think thats good, but we do need to warn that the configuration API exchanges sensitive info with the server.

Reply:
No we are not data is already decrypted by GetConfig() @donatello